### PR TITLE
Migrate `alpha-value-notation` rule to ESM

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Ref #5291

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/index.mjs [hunk 1]
```diff
@@ -6,7 +6,7 @@ import importLazy from 'import-lazy';
 
 /** @type {import('stylelint')['rules']} */
 const rules = {
-	'alpha-value-notation': importLazy(() => require('./alpha-value-notation/index.js'))(),
+	'alpha-value-notation': importLazy(() => require('./alpha-value-notation/index.cjs'))(),
```
Comment (Mouvedia) [inline lib/rules/index.mjs:9]: question

Is this a typo? Don't you want `mjs`?

Comment (ybiquitous) [reply lib/rules/index.mjs:9]: Not a typo. We still need `.cjs` when using `require()`; if using requiring `.cjs`, then the following error occurs:

```
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/masafumi.koba/git/stylelint/stylelint/lib/rules/alpha-value-notation/index.mjs not supported.
```

Like #7184, if we could replace `import-lazy` with a Promise-based code, we could rewrite `.cjs` to `.mjs` in `rules/index.mjs`.

**Review by mattxwang — APPROVED** [review]
Ah! Didn't realize you were waiting on a review, sorry.

This looks good to me! Thanks 😊

**Comment by ybiquitous** [comment]
Never mind, thanks for the review!

By the way, the coverage degradation is due to `jest-preset-stylelint` uses `require` instead of `import`. We need to change the behavior after we finish to rewrite all rules in ESM.
https://github.com/stylelint/jest-preset-stylelint/blob/6.2.0/getTestRule.js#L18

**Comment by Mouvedia** [comment]
`getTestRule` stopped me from migrating to vitest

**Pull request merged**